dask/array/fft.py

```diff
@@ -28,31 +30,37 @@ def _fft_out_chunks(a, n, axis):
     return chunks
 
 
-def _rfft_out_chunks(a, n, axis):
+def _rfft_out_chunks(a, n, axis, rtype):
     if n is None:
         n = a.chunks[axis][0]
     chunks = list(a.chunks)
-    chunks[axis] = (n // 2 + 1,)
+    if issubclass(np.dtype(rtype).type, numbers.Real):
+        chunks[axis] = (n,)
+    else:
+        chunks[axis] = (n // 2 + 1,)
```

Comment:
Ok, let's move towards erroring out if people use SciPy's `*rfft` functions then.

FWIW other people think SciPy's `*rfft` behavior here is weird and are trying to get rid of it.

xref: https://github.com/scipy/scipy/issues/2487